File: web/features/school/data/repositories/school.repository.ts
import { schoolApi } from "../api/school-api";

export const schoolRepository = {
  getSchoolsByUser() {
    return schoolApi.getSchoolsByUser();
  },

  getLevels() {
    return schoolApi.getLevels();
  },

  createSchool(data: unknown) {
    return schoolApi.createSchool(data);
  },

  joinSchoolByCode(data: unknown) {
    return schoolApi.joinSchoolByCode(data);
  },
};

File: mobile/lib/features/main/data/repositories/school_repository.dart
import 'package:mobile/features/main/main.dart';

class SchoolRepository {
  SchoolRepository({required SchoolApi schoolApi}) : _schoolApi = schoolApi;
  final SchoolApi _schoolApi;

  Future<List<School>> getSchoolsByUser() {
    return _schoolApi.getSchoolsByUser();
  }

  Future<School> joinSchoolByCode(String code) {
    return _schoolApi.joinSchoolByCode(code);
  }
}
